experimental/LieAlgebras/src/simple_lie_algebra.jl

```diff
@@ -0,0 +1,266 @@
+@attributes mutable struct SimpleLieAlgebra{C<:RingElement} <: LieAlgebra{C}
+  #Construct a simple Lie algebra over a given ring with a given root system
+  base_ring::Ring
+  root_system::RootSystem
+  dim::Int
+  s::Vector{Symbol}
+  root_type::Tuple{Symbol,Int64}
+  struct_consts::Matrix{SRow{C}}
+
+  function SimpleLieAlgebra{C}(
+    R::Ring, S::Symbol, n::Int64; cached::Bool=true
+  ) where {C<:RingElement}
+    RS = root_system(S, n)
+    dimL = number_of_roots(RS) + length(RS.simple_roots)
+    s = [Symbol("e_$i") for i in 1:dimL]
+    st = root_system_type(RS)
+    #get the structure constants of the Lie algebra L 
+    #note that it is enoough to do this over Q, as we can later coerce the constants
+    #into the ring R
+    coeffs_iso = inv(Oscar.iso_oscar_gap(QQ))
+    LG = GAP.Globals.SimpleLieAlgebra(GAP.Obj(S), n, domain(coeffs_iso))
+    sc_table_G =
+      (
+        entry -> (entry[1], Vector{elem_type(R)}(map(coeffs_iso, entry[2])))
+      ).(
+        Matrix{Tuple{Vector{Int},Vector{GAP.Obj}}}(
+          (GAP.Globals.StructureConstantsTable(GAPWrap.Basis(LG)))[1:dimL]
+        )
+      )
+    struct_consts = Matrix{SRow{elem_type(R)}}(undef, dimL, dimL)
+    for i in 1:dimL, j in 1:dimL
+      struct_consts[i, j] = sparse_row(
+        R, Tuple{Int,elem_type(R)}[(k, R(c)) for (k, c) in zip(sc_table_G[i, j]...)]
+      )
+    end
+    return get_cached!(SimpleLieAlgebraDict, (R, RS), cached) do
+      new{C}(R, RS, dimL, s, st, struct_consts)
+    end::SimpleLieAlgebra{C}
+  end
+end
+
+const SimpleLieAlgebraDict = CacheDictType{Tuple{Ring,RootSystem},SimpleLieAlgebra}()
+
+mutable struct SimpleLieAlgebraElem{C<:RingElement} <: LieAlgebraElem{C}
+  parent::SimpleLieAlgebra{C}
+  mat::MatElem{C}
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+
+parent_type(::Type{SimpleLieAlgebraElem{C}}) where {C<:RingElement} = SimpleLieAlgebra{C}
+
+elem_type(::Type{SimpleLieAlgebra{C}}) where {C<:RingElement} = SimpleLieAlgebraElem{C}
+
+parent(x::SimpleLieAlgebraElem{C}) where {C<:RingElement} = x.parent
+
+base_ring(L::SimpleLieAlgebra{C}) where {C<:RingElement} = L.base_ring::parent_type(C)
+
+coefficient_ring(L::SimpleLieAlgebra{C}) where {C<:RingElement} =
+  L.base_ring::parent_type(C)
```

Comment:
```suggestion
coefficient_ring(L::SimpleLieAlgebra{C}) where {C<:RingElement} = L.R::parent_type(C)
```
`base_ring` has been dropped in https://github.com/oscar-system/Oscar.jl/pull/2528 for the other Lie algebra types.

Reply:
I personally like working with the base_ring option. I will add the coefficient_ring and keep the base_ring